src/components/App/userReducer.ts

```diff
@@ -1,30 +1,57 @@
-import { JWTToken } from '../../../types/JWT'
 import { UID } from '../../../types/uid'
-import { SET_TOKEN } from '../../actions/types'
-
-const initialState = {
-    user: { token: '', userId: '' },
-}
+import { SIGNIN_SUCCESS } from '../../actions/userActionsTypes'
+import { JWTToken } from '../../../types/JWT'
+import { SIGNOUT, SIGNIN_ERROR } from '../../actions/userActionsTypes'
+import produce, { Draft } from 'immer'
 
 export interface User {
     token: JWTToken
     userId: UID
+    isAnonymous: boolean
 }
+
 export interface UserState {
     user: User
+    isLoading: boolean
+    error: null | unknown
```

Comment:
(non-blocking) I think we'd rather set default values to empty string instead of having `| null` for non-initialised states

Reply:
I set to null because i had a case where the firebase error was an objet